src/app/products/pages/product-page/product-page.component.ts

```diff
@@ -1,16 +1,27 @@
 import { Component } from '@angular/core';
-import { ActivatedRoute, Params } from '@angular/router';
+import { ActivatedRoute } from '@angular/router';
+import { ProductsService } from '../../service/products.service';
+import { Product } from '../../product';
 
 @Component({
   selector: 'app-product-page',
   templateUrl: './product-page.component.html',
-  styleUrls: ['./product-page.component.scss']
+  styleUrls: ['./product-page.component.scss'],
 })
 export class ProductPageComponent {
-  id!:number;
-  constructor(private route: ActivatedRoute) { }
+  id!: string;
+  product?: Product;
+
+  constructor(
+    private route: ActivatedRoute,
+    private productsService: ProductsService
+  ) {}
 
   ngOnInit() {
-    this.id = Number(this.route.snapshot.paramMap.get('id'));
+    this.id = String(this.route.snapshot.paramMap.get('id'));
```

Comment:
s obzirom da su route params svakako string, ne treba ti konverzija. 
S druge strane, ovo treba da bude broj jer je id integer.

tako da zameni ovako:
makni sa onInit i stavi ga u potpis promenjive
id: number = +this.route.snapshot.paramMap.get('id');